contracts/src/v0.8/dev/automation/2_1/KeeperRegistry2_1.sol

```diff
@@ -461,42 +449,54 @@ contract KeeperRegistry2_1 is KeeperRegistryBase2_1, OCR2Abstract, Chainable, ER
    */
   function _validateBlockTrigger(
     uint256 upkeepId,
-    BlockTrigger memory trigger,
+    bytes memory rawTrigger,
     Upkeep memory upkeep
   ) internal returns (bool) {
-    if (trigger.blockNum < upkeep.lastPerformed) {
+    BlockTrigger memory trigger = abi.decode(rawTrigger, (BlockTrigger));
+    if (trigger.blockNum < upkeep.lastPerformedBlockNumberOrTimestamp) {
       // Can happen when another report performed this upkeep after this report was generated
-      emit StaleUpkeepReport(upkeepId);
+      emit StaleUpkeepReport(upkeepId, rawTrigger);
       return false;
     }
     if (_blockHash(trigger.blockNum) != trigger.blockHash) {
```

Comment:
I think we want to allow an absence of blockhash here. Can do it in followup

Reply:
if we omit both (like set blockNum = 0 and blockHash = 0x00...) we can effectively bypass the check